docs/design/proposals/encryption-with-vault-sa.md

```diff
@@ -0,0 +1,84 @@
+# Encrypted Volumes with a ServiceAccount per Tenant
+
+## Requirements
+
+- Hashicorp Vault with Kubernetes Auth
+- each Tenant (=Namespace) has a ServiceAccount to connect to Vault
+- each Tenant can override Vault configuration options (like `BackendPath`)
+
+## Similarities with available KMS implementations
+
+Ceph-CSI provides several KMS implementations with different functionalities.
+Two implementations use Hashicorp Vault and could be used as a base for a new
```

Comment:
@nixpanic its better to expand with the pointers here about the `implementations` referred above.

Reply:
That is expanded on below already. Repeating it here is probably not needed?